src/PSR7/Validators/BodyValidator/BodyValidator.php

```diff
@@ -65,11 +65,11 @@ public function validate(OperationAddress $addr, MessageInterface $message) : vo
 
         // Validate message body
         if (preg_match('#^multipart/.*#', $contentType)) {
-            (new MultipartValidator($mediaTypeSpecs[$contentType], $contentType))->validate($addr, $message);
+            (new MultipartValidator($mediaTypeSpec, $contentType))->validate($addr, $message);
         } elseif (preg_match('#^application/x-www-form-urlencoded$#', $contentType)) {
-            (new FormUrlencodedValidator($mediaTypeSpecs[$contentType], $contentType))->validate($addr, $message);
+            (new FormUrlencodedValidator($mediaTypeSpec, $contentType))->validate($addr, $message);
```

Comment:
In this place this doesn't play any difference, that's why tests passed this. Since `preg_match` checks the strict equality here.

Reply:
The earlier tests passed because, 

This is a server request validator, the openapi schema in the tests had media-range specified for response schema and the the response schema was never validated.

Now I moved the media-ranges to `requestBody` so that it will be validated by the server request validator